File: schneider_kalf/src/form/nodes/expressions/boolean_expressions/BooleanLiteral.ts
import ExpressionVisitor from "../../visitors/ExpressionVisitor";
import Expression from "../Expression";

export default class BooleanLiteral implements Expression {
  private value: boolean;

  constructor(value: boolean) {
    this.value = value;
  }

  accept(visitor: ExpressionVisitor): any {
    return visitor.visitBooleanLiteral(this);
  }

  getValue() {
    return this.value;
  }
}

File: Ioana_Hisham/src/main/java/ql/ast/expressions/literals/BooleanLiteral.java
package ql.ast.expressions.literals;

import ql.ast.expressions.Expression;
import ql.visitors.ExpressionVisitor;

public class BooleanLiteral extends Expression {
    private final boolean value;
    public BooleanLiteral(int lineNumber, boolean value) {
        super(lineNumber);
        this.value = value;
    }

    public boolean getValue() {
        return value;
    }

    @Override
    public <T> T accept(ExpressionVisitor<T> expressionVisitor) {
        return null;
    }
}
